src/Authenticators/TOTPAuthenticator.php

```diff
@@ -75,26 +84,26 @@ public function validateTOTP($data, $request, &$result)
             if ($result->isValid()) {
                 return $member;
             }
+        } else {
+            $result->addError(_t(self::class . '.NOMEMBER', 'Member not found'));
         }
+    }
 
-        $result->addError(_t(self::class . '.NOMEMBER', 'Member not found'));
-
-        return $result;
+    /**
+     * @param BootstrapMFALoginHandler $controller
+     * @param string $name
+     * @return TOTPForm|BootstrapMFALoginForm
```

Comment:
This could possibly benefit from being a bit tighter. I see two possibilities:

* `TOTPForm` is changed to extend `BootstrapMFALoginForm`
* We use `LoginForm` as the return hint instead